src/maliput/base/distance_router.cc

```diff
@@ -0,0 +1,218 @@
+// @param routes Vector of Routes to validate.
+// @throws common::assertion_error When one of the Routes in @p routes has end to end connectivity errors.
+void ValidateEndToEndConnectivityInRoutes(const std::vector<routing::Route>& routes) {
+  for (const auto& route : routes) {
+    const std::vector<std::string> connectivity_errors = route.ValidateEndToEndConnectivity();
+    MALIPUT_VALIDATE(connectivity_errors.empty(), AggregateRouteConnectivityErrors(connectivity_errors));
+  }
+}
+
+}  // namespace
+
+DistanceRouter::DistanceRouter(const api::RoadNetwork& road_network, double lane_s_range_tolerance)
+    : Router(), road_network_(road_network), lane_s_range_tolerance_(lane_s_range_tolerance) {
+  MALIPUT_THROW_UNLESS(lane_s_range_tolerance_ >= 0.0);
+}
+
+std::vector<routing::Route> DistanceRouter::DoComputeRoutes(
+    const api::RoadPosition& start, const api::RoadPosition& end,
+    const routing::RoutingConstraints& routing_constraints) const {
+  // Validate input parameters.
+  MALIPUT_THROW_UNLESS(start.lane != nullptr);
+  MALIPUT_THROW_UNLESS(road_network_.road_geometry()->ById().GetLane(start.lane->id()) == start.lane);
+  MALIPUT_THROW_UNLESS(end.lane != nullptr);
+  MALIPUT_THROW_UNLESS(road_network_.road_geometry()->ById().GetLane(end.lane->id()) == end.lane);
+  routing::ValidateRoutingConstraints(routing_constraints);
+  static constexpr bool kRemoveUTurns{true};
+
+  // Obtain the lane sequences that connect the start with end, i.e. the routing algorithm.
+  const std::vector<std::vector<const api::Lane*>> lane_sequences =
+      routing::FindLaneSequences(start.lane, end.lane, std::numeric_limits<double>::max(), kRemoveUTurns);
+
+  // Construct the routing::Routes and routing::Phases.
+  // TODO: lateral routing::Phase inflation.
+  const double start_s = start.pos.s();
+  const double end_s = end.pos.s();
+  std::vector<routing::Route> routes;
+  for (const std::vector<const api::Lane*>& lane_sequence : lane_sequences) {
+    if (lane_sequence.size() == 1u) {
+      const routing::Phase phase(0, lane_s_range_tolerance_, {start}, {end},
+                                 {api::LaneSRange(start.lane->id(), api::SRange(start_s, end_s))}, &road_network_);
+      routes.emplace_back(std::vector<routing::Phase>{phase}, &road_network_);
+      continue;
+    }
+
+    // Handles the case when lane_sequence has a length greater than 1.
+    std::vector<routing::Phase> phases;
+    for (int i = 0; i < static_cast<int>(lane_sequence.size()); ++i) {
+      const api::Lane* lane = lane_sequence[i];
+      if (i == 0) {
+        const std::optional<double> first_end_s = routing::DetermineEdgeS(*lane, *lane_sequence[1]);
+        MALIPUT_THROW_UNLESS(first_end_s.has_value());
+        phases.emplace_back(
+            i, lane_s_range_tolerance_, std::vector<api::RoadPosition>{start},
+            std::vector<api::RoadPosition>{api::RoadPosition(lane, api::LanePosition(*first_end_s, 0., 0.))},
```

Comment:
I notice that `api::LanePosition(*first_end_s, 0., 0.)` uses `r=0` and `h=0` here and on lines 190, 191, 200, and 201. This means `start` and `end` may have non-zero values of `r` and `h` whereas all intermediate points between phases will have `r=0` and `h=0`. Is this a simplification enabled by the fact that there are no lane changes yet, so might as well stay at `r=0, `h=0`? Once lane changes are supported, could points at phase transitions have non-zero `r` and `h`? If this is a simplification, we should document it in this method's docstring and the class docstring in `distance_router.h`.

Reply:
I would not consider it a simplification, but actually a discard of useless information. No type (I am thinking here of SRange, LaneSRange, LaneSRoute, Phase, Route and Router) in the routing interface besides the points accounts for r and h coordinate information. I understand this is because the planner is managing high level information that should not be mixed up with position and orientation which is low level network information. 
Once lane changes are supported, I expect the same treatment.
Do you think we should do something else with r and h coordinates that is slipping my mind?

Reply:
I think what confuses me is `rounting::Router::DoComputeRoutes()` takes as input two `api::RoadPosition` objects, `start` and `end`, whose `r` and `h` values may be non-zero. Is it the case that `routing::Router` supports non-zero `r` and `h` at the beginning and end of a route, but imposes that everything in between has `r=0` and `h=0`?

Or, is there an expectation that the `start` and `end` input parameters into `routing::Router::ComputeRoutes()` have `r=0`, `h=0`? https://github.com/maliput/maliput/blob/50d83cf113958453fc50061a7b03adaa57034aad/include/maliput/routing/router.h#L60-L63

Reply:
Thank you for the detailed information. What confused me is whether `Route` operates at lane-level granularity or positions-within-lane-level granularity. The API appears to support positions-within-lane-level ganularity (given the `r`, `h`, and `InertialPosition` values in both its input and output values), though the router may only operate at lane-level granularity, which I believe is the case for `DistanceRouter`.